### DISCUSSION TITLE
Find Markers of Disease

### DISCUSSION THREAD

**Discussion by cmarsac** [satijalab/seurat#4258] [Q&A] [answered]
Dear Satija Lab,

I am currently trying to use the FindMarkers() function to find the markers of a given disease state. I was able to add the disease states to the Seurat object metadata and have tried coding it as a factor and a numeric but when I set the ident.1 argument equal to the disease state it returns "Cannot find the following identities in the object:" Please advise.

**Comment by timoast** [comment]
Can you include the full code you're running?

**Reply by cmarsac** [reply]
Thank you for your reply. The code is as follows:

1 library(Seurat)
2 data<-readRDS("/counts.RDS") #sc-rnaseq counts
3 annotation<-readRDS("/annotation.RDS") #corresponding patient info, disease state, etc. for each cell
4 all.genes<-rownames(data)
5
6 seurat<-CreateSeuratObject(data)
7 seurat@meta.data[["status"]]<-annotation$status #status is a factor where status1, status2, status3, etc. represent various  disease states
8
9 seurat<-NormalizeData(seurat)
10 seurat<-FindVariableFeatures(seurat)
11
12 seurat<-ScaleData(seurat, features=all.genes)
13 seurat<-RunPCA(seurat, features=all.genes, npcs=50)
14
15 seurat<-FindNeighbors(seurat)
16 seurat<-FindClusters(seurat)
17
18 seurat<-RunTSNE(seurat, reduction="pca", set.seed(1234), check_duplicates=FALSE)
19
20 markers<-FindAllMarkers(seurat, only.pos=FALSE, min.pct=0, logfc.threshold=0, ident.1="status1")

This is the error that I have been getting: 
Error in WhichCells.Seurat(object = object, idents = ident.1) : 
  Cannot find the following identities in the object: 1

I have also tried coding status as a numeric and a character variable, as well as running seurat@active.ident<-seurat@meta.data[["status"]] before running FindAllMarkers().

**Comment by timoast** [comment, marked as answer]
`ident.1` is not a parameter in the `FindAllMarkers()` function, since the function will evaluate each set of cell identities present. The issue here is that you haven't set the active cell identities to the "status" variable that you want to group cells by. You can change the active identities using the `Idents<-` function. Alternatively, if you want to find markers for one group of cells only, you can use the `FindMarkers()` function which will let you set the `ident.1` and `group.by` parameters.

**Reply by cmarsac** [reply]
Thank you for your help. Using the Idents() function seems to have solved my problem.
